src/Shared/CoreCLRAssemblyLoader.cs

```diff
@@ -66,6 +47,12 @@ public Assembly LoadFromPath(string fullPath)
 
             lock (_guard)
             {
+                if (!_resolvingHandlerHookedUp)
+                {
+                    AssemblyLoadContext.Default.Resolving += TryResolveAssembly;
```

Comment:
Would it ever be relevant to remove this handler at some point? Like in between builds when nodereuse is true. @rainersigwald I don't remember, does node reuse work on xplat?